lib/assert.js

```diff
@@ -448,13 +448,24 @@ function getActual(block) {
   return NO_EXCEPTION_SENTINEL;
 }
 
+function checkIsPromise(obj) {
+  return isPromise(obj) ||
```

Comment:
I would recommend just checking for the presence of a `.then` function here, without doing any of the other checks. That would make it compatible with Promise libraries and the `Promise.resolve` method and Promise libraries, as well as the mental model of "anything that can be `await`-ed to get a different value".

Reply:
My intention was not to support Promises/A+ but ES6 promises. And if I read the spec correct, the check is fine as I implemented it. Only checking for `obj !== null && typeof obj.then === 'function'` would allow multiple false positives, so I would rather stick to the current check.

I can remove the `isPromise` check though as any real promise will also adhere to the later checks.

Reply:
If you only want to support ES6 promises, why do you need the checks below?

Reply:
Well, I do not only want to support native promises. Every implementation that is spec compliant should be supported out of my perspective. That is not the case for all thenables. I do not have a strong opinion on this though, so I can go for what ever is suggested. Only having the upper check is definitely not enough though. The very least that should also be checked for is `typeof obj === 'object' || typeof obj === 'function` and I would argue also `typeof obj.catch === 'function'. In that case we should be relatively safe.

The difference with that check to the current implementation would be to also accept thenables that use a function instead a object as `Promise`.

Reply:
I'm not against the current check, but it would be nice to have a small comment explaining this in the code.